### PR TITLE
fix(cirriculum) - more correct solutions step 123 rpg project

### LINKED ISSUE #52962: Change the editable region and update directions for step 123 RPG game project
### Describe the Issue

On the [forum](https://forum.freecodecamp.org/search?q=step%20123), we have a lot of campers confused on which line to edit for step [123](https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-basic-javascript-by-building-a-role-playing-game/step-123) of the RPG game project. 

We should update the editable region to make it more clear where they should write their answer.
We should probably also update the directions to say to add to the existing `monsterHealth` variable 

## suggested updated change in text

```md
Following this pattern, use the addition operator (`+`) to add a random number between `1` and the value of `xp` to your `monsterHealth -= weapons[currentWeapon].power`.
```

## suggested change for starting editable region

```js
function attack() {
  text.innerText = "The " + monsters[fighting].name + " attacks.";
  text.innerText += " You attack it with your " + weapons[currentWeapon].name + ".";
  health -= monsters[fighting].level;
  --fcc-editable-region--
  monsterHealth -= weapons[currentWeapon].power;
  --fcc-editable-region--
}
```

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-basic-javascript-by-building-a-role-playing-game/step-123

### Your code

see explanation above

### Expected behavior

change the editable region and update description like mentioned above

### Screenshots

_No response_

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by a2937**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #52962

I really don't like how I did the test for "You should use `Math.floor()` to round the result of `Math.random() * xp` down." but it seems to work pretty well. I'd like some additional input and that modified assertion in particular.

### CODE REVIEW

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-asynchronous-programming-by-building-an-fcc-forum-leaderboard/643c565936a10109b3f59326.md [hunk 1]
```diff
@@ -16,13 +16,13 @@ Call the `showLatestPosts()` function at the end of your `try` block and pass in
 You should call `showLatestPosts()` at the end of your `try` block.
 
 ```js
-assert.match(code, /\s*console\.log\(data\);?\s*showLatestPosts\(.*\);??\s*/);
+assert.match(code, /\s*showLatestPosts\(.*\);??\s*/);
```
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-asynchronous-programming-by-building-an-fcc-forum-leaderboard/643c565936a10109b3f59326.md:19]: I am not sure what these changes have to do with the issue you linked to this PR. 
Why is the `console` statement being removed from this test?
Is this a bug?

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-asynchronous-programming-by-building-an-fcc-forum-leaderboard/643c565936a10109b3f59326.md [hunk 2]
```diff
@@ -16,13 +16,13 @@ Call the `showLatestPosts()` function at the end of your `try` block and pass in
...
 ```
 
 You should pass `data` as the argument to `showLatestsPosts()`.
 
 ```js
-assert.match(code, /\s*console\.log\(data\);?\s*showLatestPosts\(data\);?\s*/);
+assert.match(code, /\s*showLatestPosts\(data\);?\s*/);
```
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-asynchronous-programming-by-building-an-fcc-forum-leaderboard/643c565936a10109b3f59326.md:25]: same note with these changes
Is this a bug that needed to be fixed?

Comment (a2937) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-asynchronous-programming-by-building-an-fcc-forum-leaderboard/643c565936a10109b3f59326.md:25]: I think I made that commit on the wrong branch. :(

Comment (a2937) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-asynchronous-programming-by-building-an-fcc-forum-leaderboard/643c565936a10109b3f59326.md:25]: My last commit was meant for https://github.com/freeCodeCamp/freeCodeCamp/pull/52961. 

If it's acceptable, I'll revert it on this branch and put it where it needs to go.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-asynchronous-programming-by-building-an-fcc-forum-leaderboard/643c565936a10109b3f59326.md (hunk 1 above)
Comment (a2937) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-asynchronous-programming-by-building-an-fcc-forum-leaderboard/643c565936a10109b3f59326.md:19]: My last commit was meant for https://github.com/freeCodeCamp/freeCodeCamp/pull/52961.

If it's acceptable, I'll revert it on this branch and put it where it needs to go.

**Pull request closed without merging**